products/azure_rm_batchaccount/api.yaml

```diff
@@ -0,0 +1,101 @@
+--- !ruby/object:Api::Product
+name: Azure BatchAccount Management
+prefix: azure_rm_batchaccount
+versions:
+  - !ruby/object:Api::Product::Version
+    name: ga
+    base_url: NotUsedInAzure
+scopes:
+  - NotUsedInAzure
+azure_namespace: 'Microsoft.Batch'
+azure_version: '2018-12-01'
+objects:
+  - !ruby/object:Api::Resource
+    name: BatchAccount
+    api_name: Batch
+    base_url: NotUsedInAzure
+
+    azure_sdk_definition: !ruby/object:Api::Azure::SDKDefinition
+      provider_name: batch
+      read: !ruby/object:Api::Azure::SDKOperationDefinition
+        go_func_name: Get
+        request:
+          'resourceGroupName': !ruby/object:Api::Azure::SDKTypeDefinition::StringObject
+            id_portion: resourceGroupName
+            go_variable_name: resourceGroupName
+          'accountName': !ruby/object:Api::Azure::SDKTypeDefinition::StringObject
+            id_portion: accountName
+            go_variable_name: accountName
+        response:
+      delete: !ruby/object:Api::Azure::SDKOperationDefinition
+        go_func_name: Delete
+        request:
+          'resourceGroupName': !ruby/object:Api::Azure::SDKTypeDefinition::StringObject
+            id_portion: resourceGroupName
+            go_variable_name: resourceGroupName
+          'accountName': !ruby/object:Api::Azure::SDKTypeDefinition::StringObject
+            id_portion: accountName
+            go_variable_name: accountName
+        response:
+
+    description: |
+      Manage Azure BatchAccount instance.
+    parameters:
+      - !ruby/object:Api::Azure::Type::str
+        name: 'resource_group_name'
+        description: 'The name of the resource group that contains the Batch account.'
+        required: true
+        azure_sdk_references: ['resourceGroupName']
+      - !ruby/object:Api::Azure::Type::str
+        name: 'account_name'
+        description: 'A name for the Batch account which must be unique within the region. Batch account names must be between 3 and 24 characters in length and must use only numbers and lowercase letters. This name is used as part of the DNS name that is used to access the Batch service in the region in which the account is created. For example: http://accountname.region.batch.azure.com/.'
+        required: true
+        azure_sdk_references: ['accountName']
+      - !ruby/object:Api::Azure::Type::str
+        name: 'location'
+        description: 'The region in which to create the account.'
+        required: true
+        azure_sdk_references: ['/location']
+      - !ruby/object:Api::Azure::Type::dict
+        name: 'tags'
+        description: 'The user-specified tags associated with the account.'
+        required: true
+        azure_sdk_references: ['/tags']
+      - !ruby/object:Api::Azure::Type::dict

```

Comment:
I think at least we should flatten the first level properties under `properties` for all resources. We might need some investigation on whether to flatten properties of deeper levels.

So we may just remove this property (`properties`), and directly put its sub-properties here.